src/systemcmds/reboot/reboot.cpp

```diff
@@ -63,12 +67,19 @@ extern "C" __EXPORT int reboot_main(int argc, char *argv[])
 	int myoptind = 1;
 	const char *myoptarg = nullptr;
 
-	while ((ch = px4_getopt(argc, argv, "b", &myoptind, &myoptarg)) != -1) {
+	while ((ch = px4_getopt(argc, argv, "bi", &myoptind, &myoptarg)) != -1) {
 		switch (ch) {
 		case 'b':
 			to_bootloader = true;
 			break;
 
+#ifdef BOARD_HAS_ISP_BOOTLOADER
+
+		case 'i':
+			board_reset(2);
```

Comment:
Did you look at the operation that happen in the px4_reboot_request(to_bootloader) sequence? IIRC this is tied into ensuring a reboot will not corrupt the FRAM

Reply:
Ah k so that's why there's that mechanism. However the header is only a boolean see `int px4_reboot_request(bool to_bootloader, uint32_t delay_us)` What's preferred changing `bool to_bootloader` to `int request` or add a `bool to_isp` ?

Reply:
But I think this PR should be 2 PRs

Reply:
Seperate PR's or seperate commits?
Can you review this as is, if it's okay I can split it into 2 PR's.